src/Microsoft.DocAsCode.Build.Engine/HostService.cs

```diff
@@ -523,21 +524,30 @@ where condition(f)
         public void SaveIntermediateModel(string intermediateFolder, ModelManifest lmm, ModelManifest cmm)
         {
             var processor = Processor as ISupportIncrementalDocumentProcessor;
-            if (processor == null)
+            if (processor == null || intermediateFolder == null)
             {
                 return;
             }
 
             foreach (var pair in _modelLoadInfo)
             {
-                var fileName = Path.GetFileName(pair.Key.File);
+                string fileName;
+                do
+                {
+                    fileName = Path.GetRandomFileName();
+                } while (File.Exists(Path.Combine(intermediateFolder, cmm.BaseDir, fileName)));
                 if (pair.Value == LoadPhase.None)
                 {
                     if (lmm == null)
                     {
                         throw new InvalidDataException($"Full build hasn't loaded model {pair.Key.FullPath}");
                     }
-                    File.Copy(Path.Combine(intermediateFolder, lmm.BaseDir, fileName), Path.Combine(intermediateFolder, cmm.BaseDir, fileName));
+                    string lfn;
+                    if (!lmm.Models.TryGetValue(pair.Key.File, out lfn))
+                    {
+                        throw new InvalidDataException($"Last build hasn't loaded model {pair.Key.FullPath}");
+                    }
```

Comment:
Message is different.

---
In reply to: [78681597](https://github.com/dotnet/docfx/pull/698#discussion_r78681597) [](ancestors = 78681597)